pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -192,3 +194,20 @@ def setup_optimizers_in_pre_dispatch(self) -> bool:
         Returns: If True, delay setup optimizers till pre_dispatch, else call within setup.
         """
         return False
+
+    def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: str) -> None:
+        # dump states as a checkpoint dictionary object
+        if self.is_global_zero:
+            # write the checkpoint dictionary on the file
+
+            checkpoint = self.on_save(checkpoint)
+            try:
+                atomic_save(checkpoint, filepath)
```

Comment:
nit: moved the comment

```suggestion
            checkpoint = self.on_save(checkpoint)
            try:
                # write the checkpoint dictionary on the file
                atomic_save(checkpoint, filepath)
```